samples/subsys/video/capture/src/main.c

```diff
@@ -15,6 +16,48 @@ LOG_MODULE_REGISTER(main);
 
 #define VIDEO_DEV_SW "VIDEO_SW_GENERATOR"
 
+#define DISPLAY_FORMAT PIXEL_FORMAT_BGR_565
+
+#if DT_HAS_CHOSEN(zephyr_display)
+void display_setup(const struct device *const display_dev)
+{
+	struct display_capabilities capabilities;
+
+	if (!device_is_ready(display_dev)) {
+		LOG_ERR("Device %s not found", display_dev->name);
+		return;
+	}
+
+	printk("\nDisplay device: %s\n", display_dev->name);
+
+	display_get_capabilities(display_dev, &capabilities);
+
+	printk("- Capabilities:\n");
+	printk("  x_resolution = %u, y_resolution = %u, supported_pixel_formats = %u\n"
+	       "  current_pixel_format = %u, current_orientation = %u\n\n",
+	       capabilities.x_resolution, capabilities.y_resolution,
+	       capabilities.supported_pixel_formats, capabilities.current_pixel_format,
+	       capabilities.current_orientation);
+
+	display_set_pixel_format(display_dev, DISPLAY_FORMAT);
```

Comment:
What is the reasoning behind setting the format here? Some displays will likely want to use a different format than the one this sample sets- it is really that we need the camera format and display format to match, right?

I suppose maybe what we need to do is only enable the display if the camera format is RGB565? I suppose we could also enable XRGB888 via the ARGB8888 display format

Reply:
> What is the reasoning behind setting the format here? Some displays will likely want to use a different format than the one this sample sets- it is really that we need the camera format and display format to match, right?

Yes, the camera and display formats must match. Normally, it is the (human) user who needs to look at both camera and display default formats (and capabilities) then sets the format. But here, we want the sample work "out-of-the-box", for RT1170 case, the camera output format is XRGB so we need to set the display (we have many displays but here in this sample, it is rk055hdmi4ma0) to this format.

> I suppose maybe what we need to do is only enable the display if the camera format is RGB565? 

It's a big pity for me if we do that. The camera output on RT1170 is XRGB8888 and we cannot have camera frame displayed on LCD just because of the mismatch formats between camera and display ? I know that you want to be generic but the sample supports RT1064 and RT1170 as reference boards, we can always set the display format to match these cases, no ?

> I suppose we could also enable XRGB888 via the ARGB8888 display format

Yes, XRGB is more generic than ARGB and we can use both alternatively. But because there is no PIXEL_FORMAT_XRGB_8888 defined in display.h, I used the other.

Reply:
> I think it will be a big pity if we do that. The camera output on RT1170 is XRGB8888 and we cannot have camera frames displayed on LCD just because of the mismatch of the default formats between camera and display ? I know that you want to be generic but the sample supports RT1064 and RT1170 as reference boards, we can always set the display format to match these cases, no ? Also, through this, it will give some hints to people that we could "always" match the camera / display formats for their own camera / display set-up.

I think I'm missing something here. I guess the PXP will convert the XRGB8888 data to RGB565 for the LCD? *Ideally*, the display pipeline would report its format at XRGB8888 in this case. From my understanding, the application either requires that the format of the buffer that is output from the video pipeline and input to the display pipeline be the same, or the application converts the buffer to a new format manually.

Reply:
> I think I'm missing something here. I guess the PXP will convert the XRGB8888 data to RGB565 for the LCD ?

No, as you can see, the current PxP driver does nothing apart from rotation. It's the eLCDIF who can support various formats (via my PR which was merged sometimes ago).

> From my understanding, the application either requires that the format of the buffer that is output from the video pipeline and input to the display pipeline be the same,

I don't think so, as you can see, display controller (like eLCDIF) can support various formats, so application can set the display format. It is the actual situation now.

> or the application converts the buffer to a new format manually.

Yes, this is another way. Application can use software or hardware like PxP to do so (I am working on making a new PxP driver to do this).

Reply:
> No, as you can see, the current PxP driver does nothing apart from rotation. It's the eLCDIF itself who can support XRGB format (via https://github.com/zephyrproject-rtos/zephyr/pull/69250 which was merged sometimes ago).

Sure- sorry, I remembered seeing something for this, but wasn't sure where we added the support.

> I don't think so, as you can see, display controller (like eLCDIF) can support various formats, so application can set the display format at runtime, it does not require that the camera output format is the same as the default display format. It is the actual situation.

Yes, I follow that. To be clear, what I am asking for here is the following code:

```
/* Set display pixel format to match the one in use by the camera */
switch (capabilities.current_pixel_format) {
case VIDEO_PIX_FMT_RGB565:
    display_set_pixel_format(display_dev, PIXEL_FORMAT_BGR_565);
case VIDEO_PIX_FMT_XRGB32:
    display_set_pixel_format(display_dev, PIXEL_FORMAT_ARGB_8888);
default:
    return -ENOTSUP;
}
```

My understanding is that the Sensor->CSI pipeline on the RT1064 would output `VIDEO_PIX_FMT_RGB565` as the current format, and the Sensor->CSI2RX->CSI pipeline on the RT1170 would output `VIDEO_PIX_FMT_XRGB32` as the current format. Is that incorrect?

Reply:
Yes, this is definitely more elegant :). Thanks, done.